pkg/standalone-utils/test/GaugeActions.test.ts

```diff
@@ -0,0 +1,659 @@
+import { Account } from '@balancer-labs/v2-helpers/src/models/types/types';
+import TypesConverter from '@balancer-labs/v2-helpers/src/models/types/TypesConverter';
+
+describe('GaugeActions', function () {
+  let vault: Vault;
+  let relayer: Contract, relayerLibrary: Contract;
+  let admin: SignerWithAddress, userSender: SignerWithAddress, other: SignerWithAddress;
+
+  let adaptor: Contract, gaugeController: Contract, balMinter: Contract;
+  let BAL: Contract, veBAL: Contract, rewardToken: Contract, lpToken: Contract;
+
+  let gaugeFactory: Contract;
+  let gauge: Contract;
+
+  const totalLpTokens = fp(100);
+
+  before('get signers', async () => {
+    [, admin, userSender, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy token mocks', async () => {
+    vault = await Vault.create({ admin });
+
+    BAL = await deploy('v2-liquidity-mining/TestBalancerToken', {
+      args: [admin.address, 'Balancer', 'BAL'],
+    });
+
+    veBAL = await deploy('v2-liquidity-mining/TestBalancerToken', {
+      args: [admin.address, 'Balancer Voting Escrow', 'veBAL'],
+    });
+
+    rewardToken = await deploy('v2-liquidity-mining/TestBalancerToken', {
+      args: [admin.address, 'Reward', 'RWT'],
+    });
+
+    lpToken = await deploy('v2-pool-utils/MockBalancerPoolToken', {
+      args: ['Mock Balancer Pool Token', 'BPT', vault.address],
+    });
+
+    await lpToken.mint(userSender.address, totalLpTokens);
+  });
+
+  sharedBeforeEach('set up relayer', async () => {
+    adaptor = await deploy('v2-liquidity-mining/AuthorizerAdaptor', { args: [vault.address] });
+
+    gaugeController = await deploy('v2-liquidity-mining/MockGaugeController', {
+      args: [veBAL.address, adaptor.address],
+    });
+
+    const balTokenAdmin = await deploy('v2-liquidity-mining/MockBalancerTokenAdmin', {
+      args: [vault.address, BAL.address],
+    });
+    await BAL.connect(admin).grantRole(await BAL.MINTER_ROLE(), balTokenAdmin.address);
+
+    balMinter = await deploy('v2-liquidity-mining/BalancerMinter', {
+      args: [balTokenAdmin.address, gaugeController.address],
+    });
+
+    // Deploy Relayer: vault and BAL minter are required; we can skip wstETH.
+    relayerLibrary = await deploy('MockBatchRelayerLibrary', {
+      args: [vault.address, ZERO_ADDRESS, balMinter.address],
+    });
+    relayer = await deployedAt('BalancerRelayer', await relayerLibrary.getEntrypoint());
+
+    // Authorize Relayer for all actions
+    const relayerActionIds = await Promise.all(
+      ['setRelayerApproval', 'manageUserBalance'].map((action) => actionId(vault.instance, action))
+    );
+    await vault.grantPermissionsGlobally(relayerActionIds, relayer);
+
+    // Approve relayer by BPT holder
+    await vault.setRelayerApproval(userSender, relayer, true);
+  });
+
+  // We won't be needing the adder; we just need a staking liquidity gauge where tokens can be deposited / withdrawn.
```

Comment:
```suggestion
```

Reply:
Took me a while to understand what 'adder' meant, and by the time I realized it was the GaugeAdder I was like 'uh, yea, sure'. I don't think we need to mention this at all :shrug: